int8 enable
---
bool success
